[GTK][WPE][Skia] Remove obsolete fencing related methods from ImageBuffer/NativeImage & their backends

nikolaszimmermann opened the pull request:
#### f2876d34b120e11098f4190b4a3189c4548a0271
<pre>
[GTK][WPE][Skia] Remove obsolete fencing related methods from ImageBuffer/NativeImage &amp; their backends
<a href="https://bugs.webkit.org/show_bug.cgi?id=290283">https://bugs.webkit.org/show_bug.cgi?id=290283</a>

Reviewed by Carlos Garcia Campos.

Remove Skia specific methods from ImageBuffer / NativeImage, that
got obsoleted by the switch from DisplayList to SkPictureRecorder, for
the WPE/Gtk ports, namely:

- bool finishAcceleratedRenderingAndCreateFence()
- void waitForAcceleratedRenderingFenceCompletion()
- const GrDirectContext* skiaGrContext() const
- RefPtr&lt;ImageBuffer&gt; copyAcceleratedImageBufferBorrowingBackendRenderTarget() const

No change in behavior, just removing obsolete code.

* Source/WebCore/platform/graphics/ImageBuffer.cpp:
(WebCore::ImageBuffer::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::ImageBuffer::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::ImageBuffer::skiaGrContext const): Deleted.
(WebCore::ImageBuffer::copyAcceleratedImageBufferBorrowingBackendRenderTarget const): Deleted.
* Source/WebCore/platform/graphics/ImageBuffer.h:
* Source/WebCore/platform/graphics/ImageBufferBackend.cpp:
(WebCore::ImageBufferBackend::copyAcceleratedImageBufferBorrowingBackendRenderTarget const): Deleted.
* Source/WebCore/platform/graphics/ImageBufferBackend.h:
(WebCore::ImageBufferBackend::surface const):
(WebCore::ImageBufferBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::ImageBufferBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::ImageBufferBackend::skiaGrContext const): Deleted.
* Source/WebCore/platform/graphics/NativeImage.h:
(WebCore::NativeImageBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::NativeImageBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::NativeImageBackend::skiaGrContext const): Deleted.
(WebCore::NativeImageBackend::copyAcceleratedNativeImageBorrowingBackendTexture const): Deleted.
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.cpp:
(WebCore::ImageBufferSkiaAcceleratedBackend::ImageBufferSkiaAcceleratedBackend):
(WebCore::ImageBufferSkiaAcceleratedBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::ImageBufferSkiaAcceleratedBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::ImageBufferSkiaAcceleratedBackend::copyAcceleratedImageBufferBorrowingBackendRenderTarget const): Deleted.
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.h:
* Source/WebCore/platform/graphics/skia/NativeImageSkia.cpp:
(WebCore::PlatformImageNativeImageBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::PlatformImageNativeImageBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::PlatformImageNativeImageBackend::skiaGrContext const): Deleted.
(WebCore::PlatformImageNativeImageBackend::copyAcceleratedNativeImageBorrowingBackendTexture const): Deleted.

Canonical link: <a href="https://commits.webkit.org/292701@main">https://commits.webkit.org/292701@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/83c20725dd611e66859c77332d3c38edb82ba528

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96848 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16462 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6646 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101921 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47368 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/98893 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16758 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24907 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/73797 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31008 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99851 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/12620 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87606 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54131 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12380 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5425 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46696 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82475 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/5505 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103945 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23916 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/17450 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/82851 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/24167 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83679 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/82241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20658 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26899 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4445 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/17412 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23878 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29033 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/23537 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27017 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25278 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap commented on Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
```diff
@@ -1010,35 +999,103 @@ void GraphicsContextSkia::drawSkiaPattern(const sk_sp<SkImage>& image, const Int
+        } else {
+            auto dstRect = SkRect::MakeWH(tileRect.width(), tileRect.height());
+            auto clipRect = SkRect::MakeWH(tileFloatRectWithSpacing.width(), tileFloatRectWithSpacing.height());
+            // For raster images we can save creating a copy alltogether, by using SkPicture recording instead.
```
```suggestion
            // For raster images we can save creating a copy altogether, by using SkPicture recording instead.
```

spenap commented on Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
```diff
@@ -1010,35 +999,103 @@ void GraphicsContextSkia::drawSkiaPattern(const sk_sp<SkImage>& image, const Int
+{
+    if (!image || !image->isTextureBacked())
+        return nullptr;
+    return createAcceleratedRenderingFence<decltype(image)>(image);
```
why is `decltype` used here rather than the actual type, since we know it?

nikolaszimmermann replied:
Leftover from a previous templatified version, good catch.

nikolaszimmermann replied:
Note, your comment applies to the PR #42843, this PR just happens to contain the same commit as pre-requisite, will fix your comment in the other PR, and rebase this one, once the other one landed.

nikolaszimmermann committed:
[GTK][WPE][Skia] Remove obsolete fencing related methods from ImageBuffer/NativeImage & their backends
https://bugs.webkit.org/show_bug.cgi?id=290283

Reviewed by Carlos Garcia Campos.

Remove Skia specific methods from ImageBuffer / NativeImage, that
got obsoleted by the switch from DisplayList to SkPictureRecorder, for
the WPE/Gtk ports, namely:

- bool finishAcceleratedRenderingAndCreateFence()
- void waitForAcceleratedRenderingFenceCompletion()
- const GrDirectContext* skiaGrContext() const
- RefPtr<ImageBuffer> copyAcceleratedImageBufferBorrowingBackendRenderTarget() const

No change in behavior, just removing obsolete code.

* Source/WebCore/platform/graphics/ImageBuffer.cpp:
(WebCore::ImageBuffer::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::ImageBuffer::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::ImageBuffer::skiaGrContext const): Deleted.
(WebCore::ImageBuffer::copyAcceleratedImageBufferBorrowingBackendRenderTarget const): Deleted.
* Source/WebCore/platform/graphics/ImageBuffer.h:
* Source/WebCore/platform/graphics/ImageBufferBackend.cpp:
(WebCore::ImageBufferBackend::copyAcceleratedImageBufferBorrowingBackendRenderTarget const): Deleted.
* Source/WebCore/platform/graphics/ImageBufferBackend.h:
(WebCore::ImageBufferBackend::surface const):
(WebCore::ImageBufferBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::ImageBufferBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::ImageBufferBackend::skiaGrContext const): Deleted.
* Source/WebCore/platform/graphics/NativeImage.h:
(WebCore::NativeImageBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::NativeImageBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::NativeImageBackend::skiaGrContext const): Deleted.
(WebCore::NativeImageBackend::copyAcceleratedNativeImageBorrowingBackendTexture const): Deleted.
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.cpp:
(WebCore::ImageBufferSkiaAcceleratedBackend::ImageBufferSkiaAcceleratedBackend):
(WebCore::ImageBufferSkiaAcceleratedBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::ImageBufferSkiaAcceleratedBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::ImageBufferSkiaAcceleratedBackend::copyAcceleratedImageBufferBorrowingBackendRenderTarget const): Deleted.
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.h:
* Source/WebCore/platform/graphics/skia/NativeImageSkia.cpp:
(WebCore::PlatformImageNativeImageBackend::finishAcceleratedRenderingAndCreateFence): Deleted.
(WebCore::PlatformImageNativeImageBackend::waitForAcceleratedRenderingFenceCompletion): Deleted.
(WebCore::PlatformImageNativeImageBackend::skiaGrContext const): Deleted.
(WebCore::PlatformImageNativeImageBackend::copyAcceleratedNativeImageBorrowingBackendTexture const): Deleted.

Canonical link: https://commits.webkit.org/292701@main
